tools/logger/convert.c

```diff
@@ -170,6 +172,20 @@ static const char *asprintf_uuid(const char *fmt, uint32_t uuid_key, bool use_co
 	return format_uid(uuid_key, use_colors, be, upper);
 }
 
+static const char *asprintf_entry_text(uint32_t entry_address)
+{
+	struct ldc_entry entry;
+	int ret;
+
+	ret = read_entry_from_ldc_file(&entry, entry_address);
+	if (ret)
+		return NULL;
+
+	free(entry.file_name);
```

Comment:
Where is this allocated ? We should be freeing in the same function as we allocated for ease of debug.

Reply:
Would've been easier to chuck a c++ destructor onto it and be done with it ;)
read_entry_from_ldc_file() function allocates this field, along with entry text which we want to display.

Reply:
It would be better to free it in the allocator function, make no sense free it here.

Reply:
@lgirdwood The point is, that in other use cases, we want the file_name field populated. I suppose we might write a 'destructor' function, that would free fields of ldc_entry struct.